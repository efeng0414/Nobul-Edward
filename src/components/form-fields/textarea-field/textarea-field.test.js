import React from "react";
import renderer from "react-test-renderer";
import { IntlProvider } from "react-intl";
import TextAreaFieldWrapped from "./index";

const { WrappedComponent: TextAreaField } = TextAreaFieldWrapped;

describe("TextAreaField Component", () => {
  const componentProps = {
    intl: global.getIntl(),
    form: {
      getFieldDecorator: config => jsx => jsx,
      setFieldsValue: () => {}
    },
    id: "Testing",
    initialValue: "TESTING",
    isTextArea: true
  };

  const componentJSX = <TextAreaField {...componentProps} />;

  describe("Component Snapshot", () => {
    const snapshotJSX = (
      <IntlProvider {...global.getIntl()}>{componentJSX}</IntlProvider>
    );

    it("should match stored snapshot", () => {
      const componentSnapshot = renderer.create(snapshotJSX).toJSON();

      expect(componentSnapshot).toMatchSnapshot();
    });
  });

  describe("Component Render", () => {
    const componentMount = global.Enzyme.mount(
      componentJSX,
      global.getIntlContext({ intl: componentProps.intl })
    );

    describe("check constraints on TextAreaField component", () => {
      it("should callback with an error message if value contains email(s)", () => {
        const mockValue = "Contact me directly at efeng@nobul.com";

        const watchObject = {
          errorMessageReceived: null
        };

        const mockCallback = errorMessage => {
          watchObject.errorMessageReceived = errorMessage;
        };

        componentMount
          .instance()
          .checkCustomConstraints(null, mockValue, mockCallback);

        expect(watchObject.errorMessageReceived).toBeTruthy();
      });

      it("should callback with an error message if value contains phone number(s)", () => {
        const mockValue = "Contact me directly at 123-123-1234";

        const watchObject = {
          errorMessageReceived: null
        };

        const mockCallback = errorMessage => {
          watchObject.errorMessageReceived = errorMessage;
        };

        componentMount
          .instance()
          .checkCustomConstraints(null, mockValue, mockCallback);

        expect(watchObject.errorMessageReceived).toBeTruthy();
      });

      it("should callback with an error message if value contains hyperlink(s)", () => {
        const mockValue = "Contact me at my website: edwardRealtor.ca";

        const watchObject = {
          errorMessageReceived: null
        };

        const mockCallback = errorMessage => {
          watchObject.errorMessageReceived = errorMessage;
        };

        componentMount
          .instance()
          .checkCustomConstraints(null, mockValue, mockCallback);

        expect(watchObject.errorMessageReceived).toBeTruthy();
      });

      it("should callback with an error message if value contains MLS number", () => {
        const mockValue = "Check out my other listing: E1234567";

        const watchObject = {
          errorMessageReceived: null
        };

        const mockCallback = errorMessage => {
          watchObject.errorMessageReceived = errorMessage;
        };

        componentMount
          .instance()
          .checkCustomConstraints(null, mockValue, mockCallback);

        expect(watchObject.errorMessageReceived).toBeTruthy();
      });

      it("should not callback with an error message if value passes all constraints", () => {
        const mockValue =
          "I've sold $15 million in total volume. Number #1 agent in Toronto 5 years straight";

        const watchObject = {
          errorMessageReceived: null
        };

        const mockCallback = errorMessage => {
          watchObject.errorMessageReceived = errorMessage;
        };

        componentMount
          .instance()
          .checkCustomConstraints(null, mockValue, mockCallback);

        expect(watchObject.errorMessageReceived).toBe(undefined);
      });

      it("should clean itself if value contains inappropriate words", () => {
        const mockValue = "Accept my offer, asshole";

        const watchObject = {
          newMessageReceived: null
        };

        let mockFn = jest.fn(values => {
          watchObject.errorMessageReceived =
            values[componentMount.instance().props.id];
        });

        componentMount.setProps({
          form: {
            getFieldDecorator: config => jsx => jsx,
            setFieldsValue: mockFn
          }
        });

        componentMount.instance().checkBadWords(null, mockValue, () => {});

        expect(mockFn).toHaveBeenCalled();
        //mockValue (inappropriate) vs. newMessageReceived (clean)
        expect(watchObject.newMessageReceived).not.toEqual(mockValue);
      });
    });
  });
});
