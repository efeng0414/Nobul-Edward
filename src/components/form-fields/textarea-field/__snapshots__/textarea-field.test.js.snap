// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaField Component Component Snapshot should match stored snapshot 1`] = `
<div
  className="ant-row ant-form-item ant-form-item-no-colon"
>
  <div
    className="ant-form-item-control-wrapper"
  >
    <div
      className="ant-form-item-control"
    >
      <span
        className="ant-form-item-children"
      >
        <textarea
          className="ant-input"
          maxLength={100}
          onChange={[Function]}
          onKeyDown={[Function]}
          rows={4}
          style={Object {}}
        />
      </span>
    </div>
  </div>
</div>
`;
